src/event.rs

```diff
@@ -54,16 +58,35 @@ impl<TYPES: NodeType> LeafInfo<TYPES> {
 /// The chain of decided leaves with its corresponding state and VID info.
 pub type LeafChain<TYPES> = Vec<LeafInfo<TYPES>>;
 
+pub mod error_adaptor {
+    use super::*;
+    use serde::{de::Deserializer, ser::Serializer};
+    pub fn serialize<S: Serializer, TYPES: NodeType>(
+        elem: &Arc<HotShotError<TYPES>>,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        serializer.serialize_str(&format!("{}", elem))
+    }
+
+    pub fn deserialize<'de, D: Deserializer<'de>, TYPES: NodeType>(
+        deserializer: D,
+    ) -> Result<Arc<HotShotError<TYPES>>, D::Error> {
+        let str = String::deserialize(deserializer)?;
+        Ok(Arc::new(HotShotError::Misc { context: str }))
+    }
+}
```

Comment:
I guess `HotShotError` can't be serialized?

Reply:
Yes, we tried to do the inner dependencies serde support, but found it harder. This is workaround to convert Serialise into String and Deserialize back into String.

Reply:
Specifically, HotShotError encapsulates several external error types, two of which are dyn...

Errors on Rust are not great, in terms of ergonomics and consistency.